packages/core/src/waitFor.ts

```diff
@@ -57,13 +65,19 @@ export function waitFor<TActorRef extends AnyActorRef>(
         ? undefined
         : setTimeout(() => {
             sub!.unsubscribe();
+            if (abortListener) {
+              signal!.removeEventListener('abort', abortListener);
+            }
```

Comment:
replacing those 4 lines with `dispose()` should be fine here from what I can tell